controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -0,0 +1,651 @@
+	statusError := r.Client.Status().Update(r.ctx, r.StorageClassRequest)
+	if statusError != nil {
+		r.log.Info("Failed to update StorageClassRequest status.")
+	}
+
+	// Reconcile errors have higher priority than status update errors
+	if reconcileError != nil {
+		return result, reconcileError
+	}
+
+	if statusError != nil {
+		return result, statusError
+	}
+
+	return result, nil
+}
+
+func (r *StorageClassRequestReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueStorageConsumerRequest := handler.EnqueueRequestsFromMapFunc(
+		func(obj client.Object) []reconcile.Request {
+			annotations := obj.GetAnnotations()
+			if annotation, found := annotations[v1alpha1.StorageClassRequestAnnotation]; found {
+				parts := strings.Split(annotation, "/")
+				return []reconcile.Request{{
+					NamespacedName: types.NamespacedName{
+						Namespace: parts[0],
+						Name:      parts[1],
+					},
+				}}
+			}
+			return []reconcile.Request{}
+		})
+	// As we are not setting the Controller OwnerReference on the ceph
+	// resources we are creating as part of the StorageClassRequest, we need to
+	// set IsController to false to get Reconcile Request of StorageClassRequest
+	// for the owned ceph resources updates.
+	enqueueForNonControllerOwner := &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.StorageClassRequest{}, IsController: false}
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.StorageClassRequest{}, builder.WithPredicates(
+			predicate.GenerationChangedPredicate{},
+		)).
+		Watches(&source.Kind{Type: &rookCephv1.CephBlockPool{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &rookCephv1.CephFilesystemSubVolumeGroup{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &rookCephv1.CephClient{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &storagev1.StorageClass{}}, enqueueStorageConsumerRequest).
+		Watches(&source.Kind{Type: &snapapi.VolumeSnapshotClass{}}, enqueueStorageConsumerRequest).
+		Complete(r)
+}
+
+func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, error) {
+	r.log.Info("Running StorageClassRequest controller in Converged/Provider Mode")
+
+	r.StorageClassRequest.Status.Phase = v1alpha1.StorageClassRequestInitializing
+
+	gvk, err := apiutil.GVKForObject(&v1alpha1.StorageConsumer{}, r.Client.Scheme())
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get gvk for consumer  %w", err)
+	}
+	// reading storageConsumer Name from StorageClassRequest ownerReferences
+	ownerRefs := r.StorageClassRequest.GetOwnerReferences()
+	for i := range ownerRefs {
+		if ownerRefs[i].Kind == gvk.Kind {
+			r.storageConsumer = &v1alpha1.StorageConsumer{}
+			r.storageConsumer.Name = ownerRefs[i].Name
+			r.storageConsumer.Namespace = r.OperatorNamespace
+			break
+		}
+	}
+	if r.storageConsumer == nil {
+		return reconcile.Result{}, fmt.Errorf("no storage consumer owner ref on the storage class claim")
```

Comment:
need to update the error message to replace storageclassclaim with storageclassrequest